### PR TITLE
Add Azure Blob Storage option for the output

### LINKED ISSUE #49: Support for Azure Blob Storage
I would like to be able to store reports in Azure Blob Storage.

### PR DESCRIPTION
**Pull request by j-fuentes**
This adds a new option for the output, Azure Blob Storage (`azblob`).

Closes #49.

Signed-off-by: Jose Fuentes <jsfuentescastillo@gmail.com>

### CODE REVIEW

File: examples/aks.preflight.yaml [hunk 1]
```diff
@@ -16,9 +16,9 @@ enabled-packages:
 
 # This configures how the results will be reported.
 outputs:
-# writes a human readable doc that can be opened with a browser.
-- type: local
-  path: ./output
+# uploads a json file to Azure Blob Storage
+- type: azblobb
```
Comment (charlieegan3) [inline examples/aks.preflight.yaml]: `azblobb` -> `azblob`?

Comment (charlieegan3) [reply examples/aks.preflight.yaml]:
```suggestion
- type: azblob
```

Comment (j-fuentes) [reply examples/aks.preflight.yaml]: good catch!

**Comment by charlieegan3** [comment]
/approve

I have not tested myself but this looks simple enough.

File: cmd/check.go [hunk 2]
```diff
@@ -243,14 +244,43 @@ func check() {
...
+		} else if outputType == "azblob" {
+			outputFormat, ok := outputDefinition["format"].(string)
+			if !ok {
+				log.Fatal("Missing 'format' property in azblob output configuration.")
+			}
+			outputContainer, ok := outputDefinition["container"].(string)
+			if !ok {
+				log.Fatal("Missing 'container' property in azblob output configuration.")
+			}
+			accountName, accountKey := os.Getenv("AZURE_STORAGE_ACCOUNT"), os.Getenv("AZURE_STORAGE_ACCESS_KEY")
```
Comment (wwwil) [inline cmd/check.go]: Can we use a key file here like we do for GKE/AKS access?

Comment (j-fuentes) [reply cmd/check.go]: I looked into that briefly and didn't found an immediate well-known way of doing that, so I decided to use the env vars.

I can investigate that later.

PD: the token we use for AKS is not ideal. It is very short-lived and I haven't found a good way to extend its life. That is another thing to look into.

I have opened another issue to improve how we authenticate against Azure in general: #56

File: examples/aks.preflight.yaml [hunk 3]
```diff
@@ -16,9 +16,9 @@ enabled-packages:
 
 # This configures how the results will be reported.
 outputs:
-# writes a human readable doc that can be opened with a browser.
-- type: local
-  path: ./output
+# uploads a json file to Azure Blob Storage
```
Comment (wwwil) [inline examples/aks.preflight.yaml]: Could you also add a note here that the environment variables have to be set, and link to somewhere that explains how to do that?

File: examples/aks.preflight.yaml [hunk 4]
```diff
@@ -16,9 +16,9 @@ enabled-packages:
...
+- type: azblob
+  container: mycontainer
```
Comment (wwwil) [inline examples/aks.preflight.yaml]: Possibly also note that the container must already exist.

File: examples/aks.preflight.yaml (hunk 3 above)
Comment (j-fuentes) [reply examples/aks.preflight.yaml]: Added to `./docs/configuration.md`.

File: examples/aks.preflight.yaml (hunk 4 above)
Comment (j-fuentes) [reply examples/aks.preflight.yaml]: Added to `./docs/configuration.md`.

File: docs/configuration.md [hunk 5]
```diff
@@ -107,13 +105,62 @@ Possible formats are:
...
+
+With the `cli` type output the format is optional
+and defaults to the `cli` format, for a coloured CLI formatted report.
+
+### local
+
+The `local` type output writes the results to a local directory.
+The directory must be specified with the `path` property.
+It will be created in case it does not exist.
+
```
Comment (wwwil) [inline docs/configuration.md]: Nit: Extra newline.

**Pull request merged**